### DISCUSSION TITLE
Spawn Multiple Bots Script

### DISCUSSION THREAD

**Discussion by corey-allen** [trickerer/Trinity-Bots#819] [Q&A]
Is there a good SQL script available that will spawn in multiple bots at the same time (based on bot GUID)?

**Comment by trickerer** [comment]
SQL script - probably not, spawning a bot affects both `characters` and `world` databases, plus you'll need valid coordinates, a set of valid spawn locations maybe.
A much easier way is to write a macro, you can even write a look to spawn all existing bots at once and then `.npcbot move` them or something.
If you want exactly SQL you may want to take a look into `botcommands.cpp` `HandleNpcBotSpawnCommand()`, see what SQL queries it executes.

P.S.
You can't spawn anything based on GUID, only based on ID. GUID is generated and assigned to object automatically during spawn.
